### ISSUE TITLE
emulated resin build ignores docker build-arg

### ISSUE THREAD

**Issue by jacintoArias** [resin-io/resin-cli#580]
I apologise if this behaviour is already known, in that case I think it should be (at least temporarily) indicated in the docs, as it was not trivial to realise...

A minimal example `Dockerfile.template` working example:

```
FROM resin/%%RESIN_MACHINE_NAME%%-alpine
ARG FOO="default"
RUN echo $FOO
CMD ["bash"]
```
built with: `resin build --deviceType raspberrypi3 --arch armhf --emulated` on **macos** outputs

```
[Info]    Running emulated build
[Info]    Building Dockerfile.template project
[Build]   Step 1/4 : FROM resin/raspberrypi3-alpine
[Build]    ---> a8c5f5090816
[Build]   Step 2/4 : ARG FOO="default"
[Build]    ---> 005e391def0c
[Build]   Step 3/4 : RUN echo $FOO
[Build]    ---> Running in 629cacf0b3f9
[Build]   default
[Build]    ---> 6d3478f86c42
[Build]   Removing intermediate container 629cacf0b3f9
[Build]   Step 4/4 : CMD bash
[Build]    ---> Running in ca5195ab29d6
[Build]    ---> 8671f093940b
[Build]   Removing intermediate container ca5195ab29d6
[Build]   Successfully built 8671f093940b
```

However if I run the same on AWS **t1.micro** I get:

```
[Info]    Running emulated build
[Info]    Building Dockerfile.template project
[Build]   Step 1/5 : FROM resin/raspberrypi3-alpine
[Build]    ---> a8c5f5090816
[Build]   Step 2/5 : ARG ["FOO="default""]
[Build]    ---> Running in bacd23b60d00
[Build]    ---> a83928cd24fc
[Build]   Removing intermediate container bacd23b60d00
[Build]   Step 3/5 : COPY .resin/qemu-execve /tmp/qemu-execve
[Build]    ---> 56c04afc09f3
[Build]   Removing intermediate container 850ce9e15980
[Build]   Step 4/5 : RUN echo $FOO
[Build]    ---> Running in 8263079dc91d
[Build]    ---> 027a14b3430f
[Build]   Removing intermediate container 8263079dc91d
[Build]   Step 5/5 : CMD bash
[Build]    ---> Running in 62755ef19d9f
[Build]    ---> a3dc04c51b1d
[Build]   Removing intermediate container 62755ef19d9f
[Build]   Successfully built a3dc04c51b1d
```

docker for mac natively supports building on ARM without emulation (from resin), but on AWS all my attempts of using this feature have been futile. Any hints, workarounds, future plans?

**Comment by CameronDiver** [comment]
Hey @jacintoArias, this actually turns out to be because of the behaviour in our Dockerfile parsing library. It incorrectly parses the `ARG` line into an array, which then gives the output that you've noticed.

I've PR'd a fix here: https://github.com/resin-io-modules/docker-qemu-transpose/pull/7/files and it should be merged into resin-cli fairly soon, when the module is updated.

Thanks, nice find!

**Comment by pimterry** [comment]
This has now been released in version 0.2.2 of `docker-qemu-transpose`. If you uninstall and reinstall `resin-cli` @jacintoArias this should work as you expect, since that new submodule should get picked up and installed automatically.

I'll put out a new resin-cli release that bumps the dependency to _require_ the new version of the module shortly (probably as resin-cli 6.0.1).
